languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -66,7 +66,49 @@ private void handleCall(
       args = Optional.of(host.polarListToJava(jArgs.get()));
     }
     try {
-      registerCall(attrName, args, callId, polarInstance);
+      Object instance;
+      if (polarInstance.getJSONObject("value").has("ExternalInstance")) {
+        long instanceId =
+            polarInstance
+                .getJSONObject("value")
+                .getJSONObject("ExternalInstance")
+                .getLong("instance_id");
+        instance = host.getInstance(instanceId);
+      } else {
+        instance = host.toJava(polarInstance);
+      }
```

Comment:
```suggestion
      Object instance = host.toJava(polarInstance);
```

I believe this works. Small clean-up.

Reply:
Big cleanup! Nice! I was just moving the code that was already there.